### PR TITLE
LPS-153233 and LPS-153163 Add limited update permissions in the page editor

### PR DESCRIPTION
**Pull request by lfbesada**
https://issues.liferay.com/browse/LPS-153163
https://issues.liferay.com/browse/LPS-153233

cc: @victorg1991 🦖

### CODE REVIEW

**Comment by lfbesada** [comment]
ci:test:relevant

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by lfbesada** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectHasAnyUpdatePermission.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/**
+ * @param {{ permissions: import("../../types/ActionKeys").ActionKeysMap, selectedViewportsize: string }} state
+ */
+export default function selectHasAnyUpdatePermission({permissions}) {
+	return (
+		permissions.UPDATE ||
```
Comment (sandrodw3) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectHasAnyUpdatePermission.js:20]: Same here, don't we need adding `UPDATE_LAYOUT_ADVANCED_OPTIONS` here?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectAvailablePanels.js [hunk 2]
```diff
@@ -25,7 +25,12 @@ export default function selectAvailablePanels(panels) {
 	return function ({permissions, selectedViewportSize}) {
 		const availablePanels = ['comments', 'browser'];
 
-		if (permissions.LOCKED_SEGMENTS_EXPERIMENT || !permissions.UPDATE) {
+		if (
+			permissions.LOCKED_SEGMENTS_EXPERIMENT ||
+			(!permissions.UPDATE &&
```
Comment (sandrodw3) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectAvailablePanels.js:30]: Don't we need considering `UPDATE_LAYOUT_ADVANCED_OPTIONS` here? And by the way, I know you didn't change it, but it's me or the content of the if and the else if are the same? I guess we could join them

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectHasAnyUpdatePermission.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+/**
+ * @param {{ permissions: import("../../types/ActionKeys").ActionKeysMap, selectedViewportsize: string }} state
+ */
+export default function selectHasAnyUpdatePermission({permissions}) {
+	return (
```
Comment (sandrodw3) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectHasAnyUpdatePermission.js:19]: Same here, don't we need adding UPDATE_LAYOUT_ADVANCED_OPTIONS here?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CommonStyles.js [hunk 4]
```diff
@@ -40,6 +41,12 @@ export function CommonStyles({
 		(state) => state.selectedViewportSize
 	);
 
+	const canUpdateItemStyles = useSelector(selectCanUpdateItemStyles);
```
Comment (sandrodw3) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CommonStyles.js:44]: Um, do we consider styles with role "general" as part of the styles? Maybe it's the correct approach, just wondering

**Review by sandrodw3 — COMMENTED** [review]
Looks great 👏🏽  just left some minor questions and comments, let me know what you think! 🥰

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CommonStyles.js (hunk 4 above)
Comment (victorg1991) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CommonStyles.js:44]: Yes, even though they are in general panel those are styles too :P

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectAvailablePanels.js (hunk 2 above)
Comment (victorg1991) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectAvailablePanels.js:30]: `UPDATE_LAYOUT_ADVANCED_OPTIONS` is a special type of permission, it only enables the advanced tab it doesn't do anything else. Basically, with only that permission you cannot even see the page editor 😂

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectHasAnyUpdatePermission.js (hunk 3 above)
Comment (victorg1991) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectHasAnyUpdatePermission.js:19]: Same as above :p

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectAvailablePanels.js (hunk 2 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/selectors/selectAvailablePanels.js:30]: Cool 😛

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CommonStyles.js (hunk 4 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/browser/components/page-structure/components/item-configuration-panels/CommonStyles.js:44]: I guess it makes sense 😄

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ealonso** [comment]
ci:forward

**Pull request closed without merging**